drivers/at86rf215/at86rf215_netdev.c

```diff
@@ -798,13 +839,36 @@ static void _set_idle(at86rf215_t *dev)
 /* wake up the radio thread when on ACK timeout */
 static void _start_ack_timer(at86rf215_t *dev)
 {
-    dev->ack_msg.type = NETDEV_MSG_TYPE_EVENT;
-    dev->ack_msg.sender_pid = thread_getpid();
+    dev->timer_msg.type = NETDEV_MSG_TYPE_EVENT;
+    dev->timer_msg.sender_pid = thread_getpid();
+
+    dev->timer.arg = dev;
+    dev->timer.callback = _ack_timeout_cb;
+
+    xtimer_set(&dev->timer, dev->ack_timeout_usec);
+}
+
+/* wake up the radio thread when after CSMA backoff */
+static void _start_backoff_timer(at86rf215_t *dev)
+{
+    uint32_t base;
+    at86rf215_get_random(dev, (uint8_t *)&base, sizeof(base));
+    uint8_t be = ((dev->csma_retries_max - dev->csma_retries) - 1) + dev->csma_minbe ;
+    if (be > dev->csma_maxbe){
+	    be = dev->csma_maxbe;
+    }
+
+    uint32_t csma_backoff_usec = (((uint32_t)1 << be) - 1) * dev->csma_backoff_period;
```

Comment:
```suggestion
    uint32_t csma_backoff_usec = dev->csma_backoff_period << be;
```

Wouldn't that work as well?

Reply:
yes, but `csma_backoff_usec` would have a different value, therefore it is not as defined in the standard 

be = 2, csma_backoff_period = 5
((1 << 2) - 1) * 5 = 15
5 << 2 = 20